### DISCUSSION TITLE
Is this the correct or recommended way to create a client only vuex module?

### DISCUSSION THREAD

**Discussion by DerekAlanGilbert** [nuxt/nuxt#8984] [Questions (Nuxt 2)]
// plugins/auth.js
```
import AuthModule from '@company/module'

export default ({ store }) => {
  store.registerModule( 'Auth', AuthModule)
}
```
// nuxt.config
```
plugins: [{ src: '~/plugins/auth', mode: 'client' }],
```

**Comment by danielroe** [comment]
A perfectly valid way to do it 👍 Does it work for you?

**Reply by DerekAlanGilbert** [reply]
As of right now yes! Im currently working on re-writing one of our apps in production to use nuxt instead of vue-cli to utilize ssr, but an internal auth module we share across projects is strictly browser based so I had to find a solid solution to use it or I couldn't go forward with the re-write. I'm happy this solution is community recommended. 😄
